add hooks section

technophile-04 opened the pull request:
## Description


https://github.com/scaffold-eth/scaffoldeth.io/assets/80153681/389df1a6-7c8e-49a4-b512-ba79a67deda4


Question : 

1. Should we keep it on mobile or remove it?

2. Maybe move the hooks section below "Debug contract sections", or maybe below "Components" section but below will break zig-zag patter which we are following.

Also please feel free to criticize the code logic or suggest another approach 😅, would love to iterate/revamp the logic 🙌

Fixes #9

carletex commented:
This is looking great @technophile-04 !! Thank you.

I'd make a couple of tweaks:

- As you suggested. Move it below "Components":
   - Change the Hook Section bg to the light one
   - Switch columns left->right / right->left to keep the zig zag pattern
 
- I'd like the hook section text to mention autocompletion. Something like "Designed to simplify interactions with your deployed smart contracts. These hooks are wrappers around wagmi, an easy-to-use interface with TS autocompletions for reading from, writing to, and monitoring events emitted by your smart contracts". I'm not the best with English tho haha

Also, (maybe for another PR?) I'd have a link to the docs (component section, hook section) "Check out all the available hooks" / "Check out all the components". A button might be too strong, maybe just an underlined link.

technophile-04 commented:
Love all the suggestions, and will implement them tysm 🙌 !!

Pabl0cks commented:
Awesome job @technophile-04 it's looking great! 🙌

> As you suggested. Move it below "Components":

I like that idea you both proposed, it feels like having all the "main" features or the core from SE-2 above the "_Buidl in Community_" band, and we can add future "secondary" features below that band.

> Also, (maybe for another PR?) I'd have a link to the docs (component section, hook section) "Check out all the available hooks" / "Check out all the components". A button might be too strong, maybe just an underlined link.

I like that idea a lot also, with the underlined links.

> Should we keep it on mobile or remove it?

I feel like maybe in some mobile devices it will feel too much height.  
I'll bring another idea I saw in other websites when showing code snippets on mobile. They make the snippet horizontally scrollable, maybe in our case we could let the comment grow and make the rest scrollable to not show a super big scroll.

**Scrollable example with comment scrollable too:**

![scrollable-code-comment-1-line](https://github.com/scaffold-eth/scaffoldeth.io/assets/55535804/9b0875f6-7a1d-4cd4-9236-1a43cd538f5d)


**Scrollable example with comment not scrollable:**

![scrollable-code-comment-growing](https://github.com/scaffold-eth/scaffoldeth.io/assets/55535804/951ba054-4345-41fe-90d8-562c56f266c5)

carletex commented:
> Scrollable example with comment scrollable too

As a user, I personally like this one. Thanks for the suggestion Pablo!

technophile-04 commented:
Just updated it with suggested changes, please feel free to nitpick !!

--- 

> Scrollable example with comment scrollable too:

https://github.com/scaffold-eth/scaffoldeth.io/assets/80153681/ecd72ac6-fefc-4328-b3a9-bc28a2ba1e98

Just added a `<br/>` at comment so that it breaks into two lines for small deveices and does not go too far 

Tysm Pablo for the great suggestion 🙌 !! 

> Also, (maybe for another PR?) I'd have a link to the docs (component section, hook section) "Check out all the available hooks" / "Check out all the components"

💯💯

Pabl0cks commented on packages/nextjs/components/HooksExample.tsx:
```diff
@@ -0,0 +1,70 @@
+        <p className="my-3">
+          <Red>import</Red>
+          {" { "} useScaffoldContractWrite {" } "}
+          <Red>from</Red> <Cyan>&quot; ~~/hooks/scaffold-eth &quot;</Cyan>;
```
I think there is extra space before and after the path:

```suggestion
          <Red>from</Red> <Cyan>&quot;~~/hooks/scaffold-eth&quot;</Cyan>;
```

technophile-04 replied:
Yup 🙌

Pabl0cks commented on packages/nextjs/components/HooksExample.tsx:
```diff
@@ -0,0 +1,70 @@
+          <Purple>useScaffoldContractWrite</Purple>({"{"}
+          <br />
+          <span className="inline-block ml-4">
+            <Cyan>contractName</Cyan>: <Cyan> &quot;YourContract&quot;</Cyan>,
```
One question, should `contractName:`, `functionName:`, `args:` and `blockConfirmation:` have a different color than their values? (like black for example?). 

Maybe is just a bias from my VSC coloring.

technophile-04 replied:
Yup, actually Figma had the values in a bit darker : 

![Screenshot 2023-10-18 at 12 44 21 PM](https://github.com/scaffold-eth/scaffoldeth.io/assets/80153681/34097696-abd0-4683-bacc-6518b9b1b050)

I updated it the same as Figma, like it differentiates them a bit but also looks simple and clean

![Screenshot 2023-10-18 at 12 53 31 PM](https://github.com/scaffold-eth/scaffoldeth.io/assets/80153681/acfdce85-1e87-48d9-a04c-2d36b59dff19)

even [tokyonight theme](https://github.com/folke/tokyonight.nvim) does it similarly were keys are dark and values are light

---

I don't know if having a different color will make it more overpowered, this is how it looks with black : 

![Screenshot 2023-10-18 at 11 45 46 AM](https://github.com/scaffold-eth/scaffoldeth.io/assets/80153681/08cacfce-d7c4-4e7f-ad02-d8186d6212f5)

Pabl0cks replied:
I think is cool as you've set it 🙌

Pabl0cks commented on packages/nextjs/components/HooksExample.tsx:
```diff
@@ -0,0 +1,70 @@
+            <br />
+            <span className="inline-block ml-4">
+              console.<Purple>log</Purple>(<Cyan>&quot;Transaction blockHash&quot;</Cyan>, &nbsp;txnReceipt.
+              <Cyan>blockHash</Cyan>
```
Same for this one, I expected `blockHash` to have the same color as `txnReceipt`, but maybe is just what I'm used to.

Pabl0cks commented on packages/nextjs/pages/index.tsx:
```diff
@@ -179,23 +202,27 @@ const Home: NextPage = () => {
+            </div>
+          </div>
+
+          <div className="space-y-6 flex flex-col items-center lg:pl-16 lg:pt-4">
```
First time I have opened this last version, the "*Block Explorer*" section felt a bit too close to "*Buidl in Community*" band.

Checking old version with previous zig-zag, it seemed that the text content was a bit farther from the band, maybe we could add a bit more of `pt` on `lg` devices, at least to that text content? Maybe just a personal perception tbh 😅.

Previous:

![image](https://github.com/scaffold-eth/scaffoldeth.io/assets/55535804/1ec4904f-5bc8-4106-9dbc-a82f5f574f89)

Pabl0cks commented:
> Just updated it with suggested changes, please feel free to nitpick !!

GJ Shiv, it looks really good for me 🙌. 

I've added some nitpicks, probably most of them are subjective perceptions, so feel free to ignore them and close 😊

technophile-04 commented:
Tysm Pablo!! Just added the changes 🙌

---

Also one last thing what if change title from "Custom React hooks and utils" => "Custom Wagmi hooks and utils"  ? 

Reason: 

- Like currently wagmi is kinda of hidden and don't want people to get the wrong notion that we are still using [eth-hooks](https://github.com/scaffold-eth/eth-ui) and seeing `wagmi` people might feel at home (not another extra thing to learn to use Scaffold-eth)

- Also this https://x.com/awkweb/status/1692562844619817297?s=20

Pabl0cks commented:
> Tysm Pablo!! Just added the changes 🙌

TY Shiv, great job! I see this ready to ship 🚀

I don't have enough context for the wagmi title, looks good to me, so I'll let Carlos decide ♥

carletex commented:
This is looking great @technophile-04 

Let's merge it and we can always tweak.

> Also one last thing what if change title from "Custom React hooks and utils" => "Custom Wagmi hooks and utils" ?

I like this!

Pushed it 19d582f5fc6d4363c917f9d044ac773f62525f8c
